File: src/AdventureWorks.Web.Api/ClientApp/src/app/models/data/entities/sales/special-offer-product/special-offer-product-lookup-model.ts

export interface ISpecialOfferProductLookupModel {
  specialOfferID: number;
  productID: number;
  rowguid: string;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */
}

export class SpecialOfferProductLookupModel implements ISpecialOfferProductLookupModel {
  specialOfferID: number;
  productID: number;
  rowguid: string;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */

  constructor(init?: Partial<SpecialOfferProductLookupModel>) {
    if (!!init) {
      Object.assign<SpecialOfferProductLookupModel, Partial<SpecialOfferProductLookupModel>>(this, init);
    }
  }
}

File: src/AdventureWorks.Application/DataEngine/Sales/SpecialOfferProduct/Commands/Queries/GetSpecialOfferProducts/SpecialOfferProductLookupModel.cs
using System;
using AutoMapper;
using AdventureWorks.Application.Interfaces.Mapping;
using AdventureWorks.BaseDomain.Entities.Sales;
using System.Collections.Generic;
using AdventureWorks.Common.Extensions;
using AdventureWorks.BaseDomain.Enums;
using AdventureWorks.BaseDomain.Interfaces;
using AdventureWorks.BaseDomain.CustomTypes;
using AdventureWorks.Application.DataEngine.Production.Product.Queries.GetProducts;
namespace AdventureWorks.Application.DataEngine.Sales.SpecialOfferProduct.Queries.GetSpecialOfferProducts
{
    public partial class SpecialOfferProductLookupModel : IHaveCustomMapping
    {
        public int SpecialOfferID { get; set; }
        public int ProductID { get; set; }
        public Guid Rowguid { get; set; }
        public DateTime ModifiedDate { get; set; }

        #region Navigation Properties

        // Uncomment the following line if you need child navigation properties (can cause a poor performance)
        /*  */
        // Uncomment the following line if you need parent navigation properties
        /*  */
        #endregion

        public void CreateMappings(Profile configuration)
        {
            configuration.CreateMap<Domain.Entities.Sales.SpecialOfferProduct, SpecialOfferProductLookupModel>().IgnoreMissingDestinationMembers().CompensateWithDestinationValues();
            configuration.CreateMap<SpecialOfferProductLookupModel, BaseSpecialOfferProduct>().IgnoreMissingDestinationMembers();
        }
    }
}
